add submit_for_settlement/2 method

nimish-mehta opened the pull request:
Adds support for making settlements at a later time, if the option for settlement was false while creating the charge.

sorentwo commented on lib/transaction.ex:
```diff
@@ -88,6 +88,23 @@ defmodule Braintree.Transaction do
+  Use this if `submit_for_settlement` was false while creating the charge using sale.
+
+  ## Example
+      {:ok, transaction} = Transaction.submit_for_settlement("123", %{amount: "100"})
```
Please add an extra line between the example header and the code.

sorentwo commented on lib/transaction.ex:
```diff
@@ -88,6 +88,23 @@ defmodule Braintree.Transaction do
+
+  ## Example
+      {:ok, transaction} = Transaction.submit_for_settlement("123", %{amount: "100"})
+      transaction.status # 'settling'
```
The status will be a binary, not a charlist. Use double quotes ("settling") here.

sorentwo commented on lib/transaction.ex:
```diff
@@ -88,6 +88,23 @@ defmodule Braintree.Transaction do
+      {:ok, transaction} = Transaction.submit_for_settlement("123", %{amount: "100"})
+      transaction.status # 'settling'
+  """
+  @spec submit_for_settlement(String.t, Map.t) :: {:ok, any} | {:error, Error.t}
```
The return type would be `{:ok, t} | {:error, Error.t}`, as it will always construct a transaction on success.

nimish-mehta replied:
The type t for this file is not defined. I am assuming it won't work without it. The rest of method for this module also seem to be using `any`.
If you can point me to a resource where i can get the expected values for all the fields, I can add the type spec and update other methods as well.

sorentwo commented on test/integration/transaction_test.exs:
```diff
@@ -30,6 +30,60 @@ defmodule Braintree.Integration.TransactionTest do
+    })
+    {:error, error} = Transaction.submit_for_settlement(transaction.id, %{})
+
+    assert error.message == "Cannot submit for settlement unless status is authorized."
```
Please match instead of asserting against the entire error message. Checking the full message is brittle (do we really care if they used a capital "C" in "Cannot" or had a trailing period?)

sorentwo commented on test/integration/transaction_test.exs:
```diff
@@ -30,6 +30,60 @@ defmodule Braintree.Integration.TransactionTest do
+    })
+    {:error, error} = Transaction.submit_for_settlement(transaction.id, %{amount: "101.00"})
+
+    assert error.message == "Settlement amount is too large."
```
Same thing here with the message.

sorentwo commented:
@nimish-mehta Thanks for the PR, it's very welcome. I've left a few comments for you and then it is ready to merge.

nimish-mehta commented:
@sorentwo I have updated the documentation to use double quote, and changed the tests to use `=~` instead of `==` i am assuming this is what you meant with match. I have left a comment against one of the suggested changes, please have a look and let me know how to proceed with it.

sorentwo commented:
@nimish-mehta You're absolutely right, there wasn't a type definition for it. I've added one and updated all of the signatures. You can rebase and add `{:ok, t}` now.

nimish-mehta commented:
I have rebased and updated the spec. Let me know if any other changes are needed.

sorentwo commented:
Verified as passing locally, thanks for the work and the updates.
